From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 18:00:00 +0200
Subject: stm32mp15-pinctrl: Move I2C4 to pinctrl with PD12/PD13

On the decktrix v1 custom PCB, the PMIC (STPMIC1) I2C4 bus uses
PD12 (SCL) and PD13 (SDA) instead of the default PZ4/PZ5.

PD12/PD13 belong to GPIO port D, so the pinctrl node must be under
&pinctrl (not &pinctrl_z which is only for port Z).

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 fdts/stm32mp15-pinctrl.dtsi | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/fdts/stm32mp15-pinctrl.dtsi b/fdts/stm32mp15-pinctrl.dtsi
index 70d1db140..22262587a 100644
--- a/fdts/stm32mp15-pinctrl.dtsi
+++ b/fdts/stm32mp15-pinctrl.dtsi
@@ -434,17 +434,18 @@
 				 <STM32_PINMUX('A', 12, ANALOG)>; /* OTG_FS_DP */
 		};
 	};
-};
-
-&pinctrl_z {
 	/omit-if-no-ref/
 	i2c4_pins_a: i2c4-0 {
 		pins {
-			pinmux = <STM32_PINMUX('Z', 4, AF6)>, /* I2C4_SCL */
-				 <STM32_PINMUX('Z', 5, AF6)>; /* I2C4_SDA */
+			pinmux = <STM32_PINMUX('D', 12, AF4)>, /* I2C4_SCL */
+				 <STM32_PINMUX('D', 13, AF4)>; /* I2C4_SDA */
 			bias-disable;
 			drive-open-drain;
 			slew-rate = <0>;
 		};
 	};
+
+};
+
+&pinctrl_z {
 };
-- 
2.43.0
